product_details/storage.py

```diff
@@ -108,25 +119,36 @@ def content(self, name):
         if fo:
             return text_type(fo.content)
 
-        return None
+    def dir_data(self, name):
+        qs = self.model_class.objects.filter(name__endswith='.json')
+        if name == 'versions':
+            qs = qs.exclude(name__contains='/')
+        else:
+            qs = qs.filter(name__startswith=name + '/')
 
-    @transaction.atomic
-    def update(self, name, content, last_modified):
-        from product_details.models import ProductDetailsFile
+        data = {}
+        for fo in qs:
+            try:
+                data[fo.name] = json.loads(text_type(fo.content))
+            except ValueError:
+                continue
```

Comment:
You mean you'd rather it return `None` than an empty dict? We could do that. I think they'd be about equally useless if it happened, but it might be slightly more consistent. Though an empty object is valid JSON... Hmm...